python/lsst/pipe/tasks/imageDifference.py

```diff
@@ -319,232 +346,248 @@ def run(self, sensorRef, templateIdList=None):
+                )
+                subtractedExposure = subtractRes.subtractedExposure
 
             if self.config.doWriteMatchedExp:
                 sensorRef.put(subtractRes.matchedExposure, self.config.coaddName + "Diff_matchedExp")
 
-        if self.config.doDetection:
-            self.log.info("Computing diffim PSF")
-            if subtractedExposure is None:
-                subtractedExposure = sensorRef.get(subtractedExposureName)
-
-            # Get Psf from the appropriate input image if it doesn't exist
-            if not subtractedExposure.hasPsf():
-                if self.config.convolveTemplate:
-                    subtractedExposure.setPsf(exposure.getPsf())
-                else:
-                    if templateExposure is None:
-                        template = self.getTemplate.run(exposure, sensorRef, templateIdList=templateIdList)
-                    subtractedExposure.setPsf(template.exposure.getPsf())
-
-        # If doSubtract is False, then subtractedExposure was fetched from disk (above), thus it may have
-        # already been decorrelated. Thus, we do not do decorrelation if doSubtract is False.
-        if self.config.doDecorrelation and self.config.doSubtract:
-            decorrResult = self.decorrelate.run(exposure, templateExposure,
-                                                subtractedExposure,
-                                                subtractRes.psfMatchingKernel)
-            subtractedExposure = decorrResult.correctedExposure
+            if self.config.doDetection:
+                self.log.info("Computing diffim PSF")
+                if subtractedExposure is None:
+                    subtractedExposure = sensorRef.get(subtractedExposureName)
+
+                # Get Psf from the appropriate input image if it doesn't exist
+                if not subtractedExposure.hasPsf():
+                    if self.config.convolveTemplate:
+                        subtractedExposure.setPsf(exposure.getPsf())
+                    else:
+                        if templateExposure is None:
+                            template = self.getTemplate.run(exposure, sensorRef,
+                                                            templateIdList=templateIdList)
+                        subtractedExposure.setPsf(template.exposure.getPsf())
+
+            # If doSubtract is False, then subtractedExposure was fetched from disk (above),
+            # thus it may have already been decorrelated. Thus, we do not decorrelate if
+            # doSubtract is False.
+            if self.config.doDecorrelation and self.config.doSubtract:
+                decorrResult = self.decorrelate.run(exposure, templateExposure,
+                                                    subtractedExposure,
+                                                    subtractRes.psfMatchingKernel)
+                subtractedExposure = decorrResult.correctedExposure
+
+        # END (if subtractAlgorithm == 'AL')
```

Comment:
imageDifferenceTask has been marked for refactoring for a while now.  There's a separate ticket for that.